src/test/ui/macros/format-args-temporaries-in-write.rs

```diff
@@ -1,50 +1,53 @@
-// check-fail
+// check-pass
+#![crate_type = "lib"]
 
 use std::fmt::{self, Display};
 
 struct Mutex;
 
 impl Mutex {
-    fn lock(&self) -> MutexGuard {
-        MutexGuard(self)
+    /// Dependent item with (potential) drop glue to disable NLL.
```

Comment:
Instead of rewriting the test in this manner (which I believe was intended to show a generalization of the desired behavior), I recommend keeping the existing test structure (i.e. have `fn lock` return `MutexGuard` (where the `<'a>` is elided in the return type signature), and then add your variant as a second separate structure in this test.